packages/component-base/test/virtualizer-item-height.test.js

```diff
@@ -241,3 +241,148 @@ describe('virtualizer - item height - initial render', () => {
     });
   });
 });
+
+describe('virtualizer - item height - lazy rendering - scroll to index', () => {
+  let virtualizer;
+  let renderPlaceholders;
+  let scrollTarget;
+
+  beforeEach(() => {
+    scrollTarget = fixtureSync(`
+      <div style="height: 400px;">
+        <div class="container"></div>
+      </div>
+    `);
+    const scrollContainer = scrollTarget.firstElementChild;
+
+    renderPlaceholders = true;
+
+    virtualizer = new Virtualizer({
+      createElements: (count) => Array.from({ length: count }, () => document.createElement('div')),
+      updateElement: (el, index) => {
+        el.dataset.index = index;
+        el.style.width = '100%';
+        el.textContent = renderPlaceholders ? '' : `Item ${index}`;
+      },
+      scrollTarget,
+      scrollContainer,
+    });
+
+    virtualizer.size = 1000;
+  });
+
+  [false, true].forEach((initiallyRendered) => {
+    describe(`initially rendered: ${initiallyRendered}`, () => {
+      beforeEach(async () => {
+        if (initiallyRendered) {
+          // Setup where the virtualizer has initially rendered all the items once
+          renderPlaceholders = false;
+          virtualizer.update();
+          await aTimeout(200);
+          renderPlaceholders = true;
+        }
+      });
+
+      it('should have scrolled to the correct index after placeholders are removed', async () => {
+        // Scroll to an index while the virtualizer may still be creating physical items
+        virtualizer.scrollToIndex(500);
+
+        // At this point, only placeholders are rendered.
+        // Enable actual content rendering and update.
+        renderPlaceholders = false;
+        virtualizer.update();
+        // Wait for the content to update (and resize observer to fire)
+        await aTimeout(200);
+
+        // The first visible index should be correct
+        expect(virtualizer.firstVisibleIndex).to.equal(500);
+      });
+
+      it('should scroll to the lastly requested index', async () => {
+        virtualizer.scrollToIndex(500);
+        virtualizer.scrollToIndex(600);
+
+        renderPlaceholders = false;
+        virtualizer.update();
+        await aTimeout(200);
+
+        expect(virtualizer.firstVisibleIndex).to.equal(600);
+      });
+
+      it('should not scroll to the old index', async () => {
+        virtualizer.scrollToIndex(500);
+
+        renderPlaceholders = false;
+        virtualizer.scrollToIndex(0);
+        await aTimeout(200);
+
+        expect(virtualizer.firstVisibleIndex).to.equal(0);
+      });
+
+      it('should not try to scroll to undefined', async () => {
+        const scrollToIndexSpy = sinon.spy(virtualizer.__adapter, 'scrollToIndex');
```

Comment:
I'm not sure I understand why we need to check for exactly `undefined`. Wouldn't it be enough to just check that `scrollToIndex` doesn't get called during that 200 timeout?

Reply:
Updated the test case.